embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/comms/delivery/EmbraceDeliveryCacheManager.kt

```diff
@@ -88,24 +89,30 @@ internal class EmbraceDeliveryCacheManager(
     /**
      * This method will do disk reads so do not run it on the main thread
      */
-    override fun getAllCachedSessionIds(): List<String> {
+    override fun getAllCachedSessionIds(): List<CachedSession> {
         val sessionFileIds = cacheService.normalizeCacheAndGetSessionFileIds()
         sessionFileIds.forEach { filename ->
             val values = filename.split('.')
-            if (values.size != 4) {
+            if (values.size < 4 || values.size > 5) {
```

Comment:
I found this part kind of janky when writing tests for it early. Is it possible to abstract the validation away a bit, and do more validation on the assumptions that are later made (e.g. token 2 is a timestamp, token 3 is a session ID)? Reading this, it would be nice to just see a validation method and have all the gnarly logic in there.